File: AtividadesAula/aula04/medicao_model.py
import random
import time

class ResultadoMedicao:
    """Estrutura simples para envelopar os resultados."""
    def __init__(self, lista, tempo_ms):
        self.lista = lista
        self.tempo_ms = tempo_ms

class MedicaoModel:
    def executar_rotina(self, quantidade: int, inicio: int, fim: int, aleatorio: bool) -> ResultadoMedicao:
        lista = []
        
        # Marca o tempo inicial em nanossegundos
        tempo_inicio = time.perf_counter_ns()
        
        if aleatorio:
            for _ in range(quantidade):
                # random.randint já inclui o valor do parâmetro 'fim'
                lista.append(random.randint(inicio, fim))
        else:
            for i in range(quantidade):
                lista.append(inicio + i)
                
        # Marca o tempo final
        tempo_fim = time.perf_counter_ns()
        
        # Converte a diferença para milissegundos
        tempo_ms = (tempo_fim - tempo_inicio) / 1_000_000.0
        
        return ResultadoMedicao(lista, tempo_ms)

File: AtividadesAula/aula04/MedicaoModel.java
// É A CLASSE QUE CALCULA O TEMPO E PREENCHE A LISTA DE FORMA SEQUENCIAL OU COM RANDOM
import java.util.List;
import java.util.ArrayList;
import java.util.Random;

public class MedicaoModel {

    // Estrutura para envelopar o resultado (Lista + Tempo de execução)
    public static class ResultadoMedicao {
        public List<Integer> lista;
        public double tempoMs;

        public ResultadoMedicao(List<Integer> lista, double tempoMs) {
            this.lista = lista;
            this.tempoMs = tempoMs;
        }
    }

    public ResultadoMedicao executarRotina(long quantidade, int inicio, int fim, boolean aleatorio) {
        List<Integer> lista = new ArrayList<>();
        Random gerador = new Random();
        
        long tempoInicio = System.nanoTime();
        
        if (aleatorio) {
            for (long i = 0; i < quantidade; i++) {
                lista.add(gerador.nextInt(inicio, fim + 1));
            }
        } else {
            for (int i = 0; i < quantidade; i++) {
                lista.add(inicio + i);
            }
        }
        
        long tempoFim = System.nanoTime();
        double tempoMs = (tempoFim - tempoInicio) / 1000000.0;
        
        return new ResultadoMedicao(lista, tempoMs);
    }
}
